pontoon/base/models.py

```diff
@@ -2206,25 +2206,34 @@ def for_project_locale(
             entities = entities.filter(Q(*post_filters))
 
         # Filter by search parameters
-        if search:
+        if search and search != '"':
             # https://docs.djangoproject.com/en/dev/topics/db/queries/#spanning-multi-valued-relationships
-            search_query = (search, locale.db_collation)
-            translation_matches = (
-                entities.filter(
-                    translation__string__icontains_collate=search_query,
-                    translation__locale=locale,
-                )
-                .values_list('id', flat=True)
+            # Add a double quote at the end of search if closing quote not present
+            search = (
+                '%s"' % search
+                if search.count('"') % 2
+                else search)
+            # Split search string on spaces except if between quotes.
+            search_list = re.findall('([^\"]\\S*|\".+?\")\\s*', search)
+            search_list = [s.strip('"').strip("'").strip() for s in search_list]
+            search_query_list = [(s, locale.db_collation) for s in search_list]
+            query = reduce(operator.and_, (
```

Comment:
Can we replace this with https://github.com/mozilla/pontoon/pull/888#issuecomment-372222015? I find the proposed syntax much easier to read.

Reply:
will do. However, operator is used in one more place I the same file but with some helpful comments. would it work if I add similar comments?

Reply:
You could do that, too.

And putting the second argument of the `reduce()` in a variable would help readability as well:
```python
reduce(operator.and_, something)
```